core/src/main/scala/org/apache/spark/ui/UIUtils.scala

```diff
@@ -395,4 +397,56 @@ private[spark] object UIUtils extends Logging {
     </script>
   }
 
+  /**
+   * Returns HTML rendering of a job or stage description. It will try to parse the string as HTML
+   * and make sure that it only contains anchors with relative links. Otherwise, the whole string
+   * will rendered as a simple escaped text.
+   */
```

Comment:
can you add a paragraph on the security implications here? (What happens if I put in a `<script>` tag or something, or if I reference a malicious link)